classes/Crons.php

```diff
@@ -431,7 +431,7 @@ public static function saveRepositoryFile($module, $project_id, $filename, $tabl
             #Check if we have a different path than edocs
             $path = ProjectData::getS3Path($module, $project_id);
             $storedName = $path == null ? date("YmdHis") . "_pid" . $project_id . "_" . ProjectData::getRandomIdentifier(6) . ".txt" : $filename;
-            $filePath = $path == null ? APP_PATH_TEMP . $storedName : $path . $storedName;
+            $filePath = $path == null ? $module->getSafePath(APP_PATH_TEMP . $storedName) : $path . $storedName;
```

Comment:
Was this tested?!?  It sure looks like it would crash, as the `$root` parameter must be specified for files outside the module directory.  We test all code prior to commit going forward.  Period.  If you're not sure how to test something, please ask.

Reply:
after our talk added the fix

Reply:
Looks great.  Approved!